File: example/go/config/Task/Taskextraexp.go
package task

type Taskextraexp struct {
    taskid int //任务完成条件类型（id的范围为1-100）
    extraexp int //额外奖励经验
    test1 string
    test2 string
    fielda string
    fieldb string
    fieldc string
    fieldd string
}

//getters
func (t *Taskextraexp) GetTaskid() int {
    return t.taskid
}

func (t *Taskextraexp) GetExtraexp() int {
    return t.extraexp
}

func (t *Taskextraexp) GetTest1() string {
    return t.test1
}

func (t *Taskextraexp) GetTest2() string {
    return t.test2
}

func (t *Taskextraexp) GetFielda() string {
    return t.fielda
}

func (t *Taskextraexp) GetFieldb() string {
    return t.fieldb
}

func (t *Taskextraexp) GetFieldc() string {
    return t.fieldc
}

func (t *Taskextraexp) GetFieldd() string {
    return t.fieldd
}

var all []Taskextraexp
func GetAll() []Taskextraexp {
    return all[:len(all)]:len(all)]
}

var allMap map[int]Taskextraexp
func Get(key int) (Taskextraexp,bool){
    return allMap[key]
}
        public static Taskextraexp Get(int taskid)
        {
            Taskextraexp v;
            return all.TryGetValue(taskid, out v) ? v : null;
        }

        public static List<Taskextraexp> All()
        {
            return all.OrderedValues;
        }

        public static List<Taskextraexp> Filter(Predicate<Taskextraexp> predicate)
        {
            var r = new List<Taskextraexp>();
            foreach (var e in all.OrderedValues)
            {
                if (predicate(e))
                    r.Add(e);
            }
            return r;
        }

        internal static void Initialize(Config.Stream os, Config.LoadErrors errors)
        {
            all = new Config.KeyedList<int, Taskextraexp>();
            for (var c = os.ReadInt32(); c > 0; c--) {
                var self = _create(os);
                all.Add(self.Taskid, self);
            }
        }

File: example/lua_ls_client/cfg/task/taskextraexp.lua
local cfg = require "cfg._cfgs"

---@class cfg.task.taskextraexp
---@field taskid number , 任务完成条件类型（id的范围为1-100）
---@field extraexp number , 额外奖励经验
---@field test1 string 
---@field test2 string 
---@field fielda string 
---@field fieldb string 
---@field fieldc string 
---@field fieldd string 
---@field get fun(taskid:number):cfg.task.taskextraexp
---@field all table<any,cfg.task.taskextraexp>

local this = cfg.task.taskextraexp

local mk = cfg._mk.table(this, { { 'all', 'get', 1 }, }, nil, nil, 
    'taskid', -- int, 任务完成条件类型（id的范围为1-100）
    'extraexp', -- int, 额外奖励经验
    'test1', -- str
    'test2', -- str
    'fielda', -- str
    'fieldb', -- str
    'fieldc', -- str
    'fieldd' -- str
    )

mk(1, 1000, "1", "2", "3", "cc", "ee", "dd")

return this
